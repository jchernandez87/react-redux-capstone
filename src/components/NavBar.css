@import url('https://fonts.googleapis.com/css2?family=Lato:wght@300;400;700&display=swap');

.latoLight {
  font-family: 'Lato', sans-serif;
  font-weight: 400;
}

.navBarFlex {
  display: flex;
}

.navBar {
  flex-direction: row;
  padding: 15px 10px;
  justify-content: space-between;
  align-items: center;
  background-color: #4369b2;
  color: white;
}

.link {
  justify-content: center;
  align-items: center;
}

.arrow {
  height: 20px;
  width: 20px;
  color: white;
}

.navBarText {
  margin: 0;
}

.rightIcons {
  width: 50px;
  justify-content: space-between;
}
