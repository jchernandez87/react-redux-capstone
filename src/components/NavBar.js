import { NavLink } from 'react-router-dom';
import { FaAngleLeft, FaMicrophone, FaPlusCircle } from 'react-icons/fa';
import './NavBar.css';

const NavBar = () => {
  const links = [
    {
      id: 1,
      path: '/',
      text: 'Home',
    },
  ];

  return (
    <div className="navBar navBarFlex">
      <NavLink to={links[0].path} className="link navBarFlex">
        <FaAngleLeft className="arrow" />
      </NavLink>
      <div>
        <h3 className="navBarText latoLight">covid stats</h3>
      </div>
      <div className="rightIcons navBarFlex">
        <FaMicrophone />
        <FaPlusCircle />
      </div>
    </div>
  );
};

export default NavBar;
