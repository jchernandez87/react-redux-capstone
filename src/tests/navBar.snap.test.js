import { BrowserRouter as Router } from 'react-router-dom';
import '@testing-library/jest-dom';
import renderer from 'react-test-renderer';
import NavBar from '../components/NavBar';

it('Check display navBar', () => {
  const tree = renderer
    .create(
      <Router>
        <NavBar />
      </Router>,
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
