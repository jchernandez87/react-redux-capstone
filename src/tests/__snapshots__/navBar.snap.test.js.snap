// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Check display navBar 1`] = `
<div
  className="navBar navBarFlex"
>
  <a
    aria-current="page"
    className="link navBarFlex active"
    href="/"
    onClick={[Function]}
    style={Object {}}
  >
    <svg
      className="arrow"
      fill="currentColor"
      height="1em"
      stroke="currentColor"
      strokeWidth="0"
      style={
        Object {
          "color": undefined,
        }
      }
      viewBox="0 0 256 512"
      width="1em"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M31.7 239l136-136c9.4-9.4 24.6-9.4 33.9 0l22.6 22.6c9.4 9.4 9.4 24.6 0 33.9L127.9 256l96.4 96.4c9.4 9.4 9.4 24.6 0 33.9L201.7 409c-9.4 9.4-24.6 9.4-33.9 0l-136-136c-9.5-9.4-9.5-24.6-.1-34z"
      />
    </svg>
  </a>
  <div>
    <h3
      className="navBarText latoLight"
    >
      covid stats
    </h3>
  </div>
  <div
    className="rightIcons navBarFlex"
  >
    <svg
      fill="currentColor"
      height="1em"
      stroke="currentColor"
      strokeWidth="0"
      style={
        Object {
          "color": undefined,
        }
      }
      viewBox="0 0 352 512"
      width="1em"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M176 352c53.02 0 96-42.98 96-96V96c0-53.02-42.98-96-96-96S80 42.98 80 96v160c0 53.02 42.98 96 96 96zm160-160h-16c-8.84 0-16 7.16-16 16v48c0 74.8-64.49 134.82-140.79 127.38C96.71 376.89 48 317.11 48 250.3V208c0-8.84-7.16-16-16-16H16c-8.84 0-16 7.16-16 16v40.16c0 89.64 63.97 169.55 152 181.69V464H96c-8.84 0-16 7.16-16 16v16c0 8.84 7.16 16 16 16h160c8.84 0 16-7.16 16-16v-16c0-8.84-7.16-16-16-16h-56v-33.77C285.71 418.47 352 344.9 352 256v-48c0-8.84-7.16-16-16-16z"
      />
    </svg>
    <svg
      fill="currentColor"
      height="1em"
      stroke="currentColor"
      strokeWidth="0"
      style={
        Object {
          "color": undefined,
        }
      }
      viewBox="0 0 512 512"
      width="1em"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M256 8C119 8 8 119 8 256s111 248 248 248 248-111 248-248S393 8 256 8zm144 276c0 6.6-5.4 12-12 12h-92v92c0 6.6-5.4 12-12 12h-56c-6.6 0-12-5.4-12-12v-92h-92c-6.6 0-12-5.4-12-12v-56c0-6.6 5.4-12 12-12h92v-92c0-6.6 5.4-12 12-12h56c6.6 0 12 5.4 12 12v92h92c6.6 0 12 5.4 12 12v56z"
      />
    </svg>
  </div>
</div>
`;
